core/src/test/java/org/apache/calcite/sql2rel/RelFieldTrimmerTest.java

```diff
@@ -0,0 +1,137 @@
+  public static Frameworks.ConfigBuilder config() {
+    final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+    return Frameworks.newConfigBuilder()
+        .parserConfig(SqlParser.Config.DEFAULT)
+        .defaultSchema(
+            CalciteAssert.addSchema(rootSchema, CalciteAssert.SchemaSpec.SCOTT_WITH_TEMPORAL))
+        .traitDefs((List<RelTraitDef>) null)
+        .programs(Programs.heuristicJoinOrder(Programs.RULE_SET, true, 2));
+  }
+
+  @Test public void testSortExchangeFieldTrimmer() {
+    final RelBuilder builder = RelBuilder.create(config().build());
+    final RelNode root =
+        builder.scan("EMP")
+            .project(builder.field("EMPNO"), builder.field("ENAME"), builder.field("DEPTNO"))
+            .sortExchange(RelDistributions.hash(Lists.newArrayList(1)), RelCollations.of(0))
+            .project(builder.field("EMPNO"), builder.field("ENAME"))
+            .build();
+
+    System.out.println(Util.toLinux(RelOptUtil.toString(root)));
+
+    RelFieldTrimmer fieldTrimmer = new RelFieldTrimmer(null, builder);
+    RelNode trimmed = fieldTrimmer.trim(root);
+
+    final String expected = ""
+        + "LogicalSortExchange(distribution=[hash[1]], collation=[[0]])\n"
+        + "  LogicalProject(EMPNO=[$0], ENAME=[$1])\n"
+        + "    LogicalTableScan(table=[[scott, EMP]])\n";
+    assertThat(trimmed, hasTree(expected));
+  }
+
+  @Test public void testSortExchangeFieldTrimmerWhenProjectCannotBeMerged() {
+    final RelBuilder builder = RelBuilder.create(config().build());
+    final RelNode root =
+        builder.scan("EMP")
+            .project(builder.field("EMPNO"), builder.field("ENAME"), builder.field("DEPTNO"))
+            .sortExchange(RelDistributions.hash(Lists.newArrayList(1)), RelCollations.of(0))
+            .project(builder.field("EMPNO"))
+            .build();
+
+    System.out.println(Util.toLinux(RelOptUtil.toString(root)));
+
+    RelFieldTrimmer fieldTrimmer = new RelFieldTrimmer(null, builder);
+    RelNode trimmed = fieldTrimmer.trim(root);
+
+    final String expected = ""
+        + "LogicalProject(EMPNO=[$0])\n"
+        + "  LogicalSortExchange(distribution=[hash[1]], collation=[[0]])\n"
+        + "    LogicalProject(EMPNO=[$0], ENAME=[$1])\n"
+        + "      LogicalTableScan(table=[[scott, EMP]])\n";
+    assertThat(trimmed, hasTree(expected));
+  }
+
+  @Test public void testSortExchangeFieldTrimmerWithEmptyCollation() {
+    final RelBuilder builder = RelBuilder.create(config().build());
+    final RelNode root =
+        builder.scan("EMP")
+            .project(builder.field("EMPNO"), builder.field("ENAME"), builder.field("DEPTNO"))
+            .sortExchange(RelDistributions.hash(Lists.newArrayList(1)), RelCollations.EMPTY)
+            .project(builder.field("EMPNO"), builder.field("ENAME"))
+            .build();
+
+    System.out.println(Util.toLinux(RelOptUtil.toString(root)));
+
+    RelFieldTrimmer fieldTrimmer = new RelFieldTrimmer(null, builder);
```

Comment:
No need to print the plan. So I think we'd better remove `System.out.println(Util.toLinux(RelOptUtil.toString(root)));`.

Reply:
Removed.